pkg/network/tracer/connection/kprobe/tcp_close_consumer.go

```diff
@@ -42,35 +42,33 @@ func newTCPCloseConsumer(cfg *config.Config, m *manager.Manager, perfHandler *dd
 	}
 
 	numCPUs := int(connCloseEventMap.ABI().MaxEntries)
-	batchManager, err := newPerfBatchManager(connCloseMap, numCPUs, filter)
+	batchManager, err := newPerfBatchManager(connCloseMap, numCPUs)
 	if err != nil {
 		return nil, err
 	}
 
 	c := &tcpCloseConsumer{
-		perfHandler:   perfHandler,
-		batchManager:  batchManager,
-		requests:      make(chan requestPayload),
-		closedBuffer:  network.NewConnectionBuffer(1024),
-		maxBufferSize: cfg.MaxClosedConnectionsBuffered,
+		perfHandler:  perfHandler,
+		batchManager: batchManager,
+		requests:     make(chan requestPayload),
 	}
-	c.start()
-
 	return c, nil
 }
 
-func (c *tcpCloseConsumer) GetClosedConnections(buffer *network.ConnectionBuffer) {
-	if buffer == nil {
-		return
+func (c *tcpCloseConsumer) GetPendingConns() *connection.ClosedBatch {
+	if c == nil {
+		return nil
 	}
 
+	responseBatch := connection.GetBatch()
 	request := requestPayload{
-		buffer:       buffer,
+		buffer:       responseBatch.Buffer,
 		responseChan: make(chan struct{}),
 	}
 
 	c.requests <- request
 	<-request.responseChan
+	return responseBatch
```

Comment:
I think `Release()` is never called on this `Batch`, and even if it was, it might not do anything because `Release` bails if the length isn't 5.

Reply:
that's actually intentional. there is not much point in keeping this specific `ConnBatch`, because it's generated once every 30s and the size can vary from 0 to TCP_BATCH_SIZE*NUM_CPUS

Reply:
On a second thought, I think I should manually instantiate this `ClosedBatch` instead of retrieving it from the pool, to make this distinction more clear. What do you think?

Reply:
I'm OK with it, either way.